src/app/(pages)/talks-and-roundtables/[slug]/page.tsx

```diff
@@ -1,27 +1,93 @@
-// import { notFound } from 'next/navigation'
-//
-// import { fetchDoc } from '../../../_api/fetchDoc'
-
-export const dynamic = 'force-dynamic'
-
-export default async function TalksAndRoundTablesPage() {
- /* let content = null
-
-  try {
-    content = await fetchDoc({
-      collection: 'talks-and-roundtables',
-      slug: slug,
-    })
-  } catch (err) {}
-
-  if (!content) {
-    notFound()
-  }
-*/
+import { fetchContentBySlug } from "@/app/_utilities/contentFetchers";
+import getVideoId from "get-video-id";
+import BackButton from "@/app/_components/BackButton";
+import ArchiveButton from "@/app/_components/ArchiveButton";
+import { formatDateTime } from "@/app/_utilities/formatDateTime";
+import Share from "@/app/_components/Share";
+import Categories from "@/app/_components/Categories";
+import AuthorPill from "@/app/_components/AuthorPill";
+import styles from "./styles.module.css";
+import SocialLinks from "@/app/_components/SocialLinks";
+import { Subscribe } from "@/app/_blocks/Subscribe";
+
+export const dynamic = "force-dynamic";
+
+export default async function TalksAndRoundTablesPage({ params: paramsPromise }) {
+  const { slug } = await paramsPromise;
+
+  const talk = await fetchContentBySlug({
+    slug: slug,
+    type: "talks-and-roundtables",
+  });
+
+  // move this to the collection / store in db?
+  // @ts-ignore
+  const videoID = getVideoId(talk.url);
+
   return (
     <div>
-      {/*Hello, world!
-      <pre>{JSON.stringify(content, null, 2)}</pre>*/}
+      {/*{<pre>{JSON.stringify(talk, null, 2)}</pre>}*/}
+      {/* Head Block */}
+      <div className={styles.headContainer}>
+        <BackButton color={"var(--soft-white-100)"} />
+        <ArchiveButton collection={"talks-and-roundtables"} color={"var(--soft-white-100)"} />
+        {/* @ts-ignore */}
+        <h5>{talk.title}</h5>
+        <p>
+          {/* @ts-ignore */}
+          {formatDateTime(talk.publishedAt)}
+          <span>Duration in mins</span>
+        </p>
+
+      </div>
+      <div className={styles.contentContainer}>
+
+        {/* Contributors column*/}
+        <div className={styles.contributors}>
+          <p className={"overline"}>
+            CONTRIBUTORS
+          </p>
+          {/* @ts-ignore */}
+          {talk.authors.map((author) => (
+            <>
+              <AuthorPill author={author} />
+              {/* TODO: Enable once most recent PR hits main */}
+              {/*<SocialLinks socials={author.socials} />*/}
+            </>
+          ))}
+        </div>
+
+        {/* Content column */}
+        <div className={styles.videoPlayer}>
+          {videoID.service === "youtube" && (
```

Comment:
What are the other service options? One option would be for people to paste the embed link in the CMS, and we just render 
```
<iframe width="560"
              height="315"
              src={talk.url}
```